src/Record.js

```diff
@@ -151,8 +151,14 @@ export class Record {
   }
 
   clear() {
+    if (this.__ownerID) {
+      this._values.clear();
+
+      return this;
+    }
```

Comment:
I inspired myself with the [the 3.8.2 implementation](https://github.com/immutable-js-oss/immutable-js/blob/v3.8.2/src/Record.js#L67-L70)

The issue here was that clearing the list of values and then calling `setSize` does set the elements to be "null" instead of "undefined.
Breaking early without calling setSize does fix the problem apparently.

Reply:
I'm wary of this change because before this change `this._values` will be always be the correct size.

My suggested alternative change would be to make it so that `List` returns `undefined` when an element isn't set (which seems to be the intention based on [the `List#setSize()` documentation](https://immutable-js.github.io/immutable-js/docs/#/List/setSize) and based on the fact that it works like that in the majority of cases). In order to accomplish this we would change [the `List#clear()` logic](https://github.com/immutable-js-oss/immutable-js/blob/master/src/List.js#L109-L110) from:
```
this._root = this._tail = null;
this.__hash = undefined;
```
to:
```
this._root = this._tail = this.__hash = undefined;
```

I'm fairly certain this won't break anything because `_root` and `_tail` start as `undefined` [when an `emptyList()` is created](https://github.com/immutable-js-oss/immutable-js/blob/master/src/List.js#L414-L417).

Reply:
@Methuselah96 you were right. This is a better approach. I added tests for both `Record` and `List.clear.setSize`